### PR TITLE
bugfix: handle allowUnfree = false

### PR DESCRIPTION
**Pull request by milahu**
nix-gui was failing with the default config `allowUnfree = false`

also nix-gui did not check the returncode of nix-instantiate
which can print some output to stderr, and then throw

### disable trace by default

mostly we need only the first error
show traces only on demand via `nix-gui --show-trace`
dont automatically "retry" nix-instantiate

### related: allowUnfree true is bad

setting `allowUnfree = true` is a bad habit, better is `allowUnfreePredicate`

per https://github.com/NixOS/nixpkgs/issues/55674#issuecomment-950350916

```nix
{
# allow ALL unfree packages
#nixpkgs.config.allowUnfree = true;

# allow some unfree packages
nixpkgs.config.allowUnfreePredicate = pkg: builtins.elem (lib.getName pkg) [
  "cnijfilter2" # canon printer
  "font-bh-lucidatypewriter-75dpi" # https://github.com/NixOS/nixpkgs/issues/99014
  "corefonts"
];
}
```

### CODE REVIEW

File: nixui/options/nix_eval.py [hunk 1]
```diff
@@ -4,11 +4,19 @@
...
 from string import Template
+import os
+import sys
 
 from nixui.utils.logger import logger
 from nixui.utils import cache
 from nixui.options.attribute import Attribute
 
+env_nix_instantiate = os.environ.copy()
+env_nix_instantiate["NIXPKGS_ALLOW_UNFREE"] = "1"
```
Comment (lapp0) [inline nixui/options/nix_eval.py:15]: Isn't `NIXPKGS_ALLOW_UNFREE` inherited from the environment in which Nix-Gui is run?

File: nixui/main.py [hunk 2]
```diff
@@ -39,6 +39,11 @@ def handle_args():
         help="Run with profiler.",
         action='store_true',
     )
+    optional.add_argument(
```
Comment (lapp0) [inline nixui/main.py:42]: my main concern here is that `--show-trace` is trivial for `nix` commands, but to actually get the traceback in Nix-Gui, users will need to reproduce their state after starting again with `--show-trace`

File: nixui/options/nix_eval.py (hunk 1 above)
Comment (milahu) [reply nixui/options/nix_eval.py:15]: yes, thats the problem. when NIXPKGS_ALLOW_UNFREE is false, the nix-instantiate command throws, and nix-gui fails to parse the invalid json

File: nixui/main.py (hunk 2 above)
Comment (milahu) [reply nixui/main.py:42]: can be solved better in a gui dialog / gui status bar message

Comment (lapp0) [reply nixui/main.py]: I think this change is risky

**Comment by lapp0** [comment]
Closed in favor of https://github.com/nix-gui/nix-gui/pull/185/files

**Pull request closed without merging**